src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -281,20 +288,20 @@ public static void AddScale(float scale, ReadOnlySpan<float> source, Span<float>
             Contracts.Assert(count <= source.Length);
             Contracts.Assert(count <= destination.Length);
 
-            if (Avx.IsSupported)
+            if (destination.Length < MinInputSize || !(Avx.IsSupported || Sse.IsSupported))
             {
-                AvxIntrinsics.AddScaleU(scale, source, destination, count);
+                for (int i = 0; i < count; i++)
```

Comment:
just a nit: Given that this is always less than 16, I imagine it would be worth checking the numbers for the unrolled version of the loop.

Reply:
sure i will run the numbers with unrolled version of the loop